jwt.js

```diff
@@ -227,6 +229,15 @@ function fastifyJwt (fastify, options, next) {
           }
           callback(err, result)
         })
+      },
+      function checkIfIsTrusted (result, callback) {
+        if (!trusted) {
+          callback(null, result)
+        } else {
+          Promise.resolve(trusted(request, result))
```

Comment:
I would change this like this: https://github.com/fastify/fastify-auth/blob/87947b90e0f21deab6cc027c570ca66693ee1888/fastify-auth.js#L73-L77

in order to avoid to create unnecessary promises

Reply:
While refactoring this piece it came to my mind how should we handle the case when the result of calling `trusted` function is neither a `boolean` nor a `Promise`. Should we throw an exception?

I think this snippet will help clarify my question:
```js
if (!trusted) {
  callback(null, result)
} else {
  const maybePromise = trusted(request, result)

  if (typeof maybePromise === 'boolean') {
    maybePromise ? callback(null, maybePromise) : callback(new Unauthorized(messagesOptions.authorizationTokenUntrusted))
  } else if (maybePromise && typeof maybePromise.then === 'function') {
    maybePromise
      .then(trusted => trusted ? callback(null, result) : callback(new Unauthorized(messagesOptions.authorizationTokenUntrusted)))
  } else {
    // TODO: how should we handle this case?
  }
}
```

Reply:
I would just assume that it is a “truthy” or “falsy” value, I would not validate its type.